builtin/logical/pki/cert_util.go

```diff
@@ -1577,27 +1576,32 @@ func generateCreationBundle(b *backend, data *inputBundle, caSign *certutil.CAIn
 }
 
 // compute a certificate's Not Before based on the role and input api data sent. Returns notBefore Time and an error.
-func getCertificateNotBefore(data *inputBundle) (time.Time, error) {
+func getCertificateNotBefore(data *inputBundle) (time.Time, []string, error) {
 	var notBefore time.Time
+	var warnings []string
 	var err error
 
+	notBeforeBound := data.role.NotBeforeBound
 	notBeforeAlt := data.role.NotBefore
 
 	if notBeforeAlt == "" {
 		notBeforeAltRaw, ok := data.apiData.GetOk("not_before")
-		if ok {
+		if ok && notBeforeBound == certutil.PermitNotBeforeBound.String() {
 			notBeforeAlt = notBeforeAltRaw.(string)
 		}
+		if ok && notBeforeBound == certutil.ForbidNotBeforeBound.String() {
+			return time.Time{}, warnings, errutil.UserError{Err: fmt.Sprintf("not_before_bound is set to %s. not_before cannot be provided.", notBeforeBound)}
+		}
 	}
```

Comment:
Somewhat personal but something like this would probably be easier to read. Thoughts?

```go
	if roleNotBefore == "" {
		apiNotBefore, hasApiNotBefore := data.apiData.GetOk("not_before")
		if hasApiNotBefore {
			switch notBeforeBound {
			case certutil.PermitNotBeforeBound.String():
				// Allow custom not-before date from API
				roleNotBefore = apiNotBefore.(string)
			case certutil.ForbidNotBeforeBound.String():
				// Reject custom not-before dates
				return time.Time{}, warnings, errutil.UserError{
					Err: fmt.Sprintf("custom not_before dates are forbidden (not_before_bound=%s). Remove the not_before parameter to proceed.", notBeforeBound),
				}
			}
		}
	}
```

Reply:
I have added a switch statement. Please let me know if you'd also like to change the variable names, etc..

Reply:
Hey,
No, that's fine. My suggestion was only regarding logic, the var names were illustrative.